core/lib/generators/solidus/install/install_generator/bundler_context.rb

```diff
@@ -49,7 +49,11 @@ def self.bundle_cleanly(&block)
       end
 
       def initialize
-        @dependencies = Bundler.locked_gems.dependencies
+        # We want to get the locked gems from `#definition` instead of
+        # `Bundler.locked_gems` so that the locked gems won't be memoized. See
+        # https://rubydoc.info/gems/bundler/2.3.22/Bundler.locked_gems.
+        @dependencies = Bundler.definition.locked_gems.dependencies
```

Comment:
Highlighting the comment about calling `locked_gems` from `Bundler.definition`:

```
        # We want to get the locked gems from `#definition` instead of
        # `Bundler.locked_gems` so that the locked gems won't be memoized. See
        # https://rubydoc.info/gems/bundler/2.3.22/Bundler.locked_gems.
```

Reply:
@kennyadsl suggested wrapping that in a `uncached_dependencies` method and moving the comment into the commit message.

Reply:
@kennyadsl @waiting-for-dev I'm not sure about the desired outcome for `uncached_dependencies`. Is our goal to extract `uncached_dependencies` in a private method, like this?

```ruby
module Solidus
  class InstallGenerator < Rails::Generators::AppBase
    def initialize
        @dependencies = uncached_dependencies
        @injector = InjectorWithPathSupport
    end

    private

    # We want to get the locked gems from `#definition` instead of
    # `Bundler.locked_gems` so that the locked gems won't be memoized. See
    # https://rubydoc.info/gems/bundler/2.3.22/Bundler.locked_gems.
    def uncached_dependencies
      Bundler.definition.locked_gems.dependencies
    end
  end
end
```

For this case, I would still prefer to keep the comment inline with the method since it's pertains specifically to the method.

Reply:
Yeah, I think that's what @kennyadsl suggested. I don't have a strong opinion about the comment 🙂